packages/form/src/cli.ts

```diff
@@ -26,17 +26,70 @@ switch (cmd) {
 async function create() {
 	intro(`Create a new form component`);
 	// Do stuff
-	const useReact = handleCancel(
-		await confirm({
-			message: "Do you want to use React?",
-			initialValue: true,
-		}),
-	);
-
-	if (!useReact) {
-		note("Well, sorry. We only support React right now ;)");
+	const frameworks = [
+		// ? this code should be here?
+		{
+			value: "react",
+			label: "React",
+			templatePath: "react/Form.tsx",
+			componentName: "Form.tsx",
+		},
+		{
+			value: "preact",
+			label: "Preact",
+			templatePath: "preact/Form.tsx",
+			componentName: "Form.tsx",
+		},
+	] as const;
+
+	type Framework = (typeof frameworks)[number];
+
+	let foundFramework: Framework | null = null;
+	const packageJsonPath = resolve(process.cwd(), "package.json");
+	if (existsSync(packageJsonPath)) {
+		const {
+			dependencies = {},
+			devDependencies = {},
+			peerDependencies = {},
+		} = JSON.parse(await readFile(packageJsonPath, { encoding: "utf-8" }));
+
+		for (const framework of frameworks) {
+			if (
+				Object.keys(dependencies).includes(framework.value) ||
+				Object.keys(devDependencies).includes(framework.value) || // ? should we check this?
+				Object.keys(peerDependencies).includes(framework.value) // ? should we check this?
+			) {
+				foundFramework = framework;
+				break;
+			}
+		}
 	}
 
+	const useFoundFramework =
+		!!foundFramework &&
+		handleCancel(
+			await confirm({
+				message: `Do you want to use ${foundFramework.label}?`, // ? What would be a better message?
```

Comment:
```suggestion
				message: `Do you want to use ${foundFramework.label}?`,
```

This looks fine to me!